src/js/components/ServiceForm.js

```diff
@@ -40,6 +40,8 @@ const FIELDS_TO_WATCH = {
   }
 };
 
+let networkTypeChanged = false;
```

Comment:
I still think it's important to ship quality for patches :) Since we're keeping a component's state outside of the component, couldn't we potentially end up with an issue after mounting/unmounting lifecycles? I just don't think this is a good practice, unless there is a strong argument for it in this case.